### PR TITLE
[LoongArch64] Remove the register tp and rewrite the definition of PTFF_SAVE_* in NativeAOT.

### PR DESCRIPTION
**Pull request by sunlijun-610**
This PR modifies:
1. Remove the incorrect usages of register tp.
2. Rewrite the definition of PTFF_SAVE_*.
3. Revise `andi+beq/bne` to `bstrpick.d+beq/bne`.

### CODE REVIEW

**Comment by sunlijun-610** [comment]
@shushanhf Could you please review this PR?
Thanks!

File: src/coreclr/nativeaot/Runtime/loongarch64/ExceptionHandling.S [hunk 1]
```diff
@@ -512,7 +512,7 @@ LOCAL_LABEL(DonePopping):
 
         PREPARE_EXTERNAL_VAR_INDIRECT_W RhpTrapThreads, $a3
 
-        andi  $t7, $a3, TrapThreadsFlags_AbortInProgress_Bit
+        bstrpick.d  $t7, $a3, TrapThreadsFlags_AbortInProgress_Bit, TrapThreadsFlags_AbortInProgress_Bit
```
Comment (shushanhf) [inline src/coreclr/nativeaot/Runtime/loongarch64/ExceptionHandling.S:515]: Please don't modify here.

File: src/coreclr/nativeaot/Runtime/loongarch64/GcProbe.S [hunk 2]
```diff
@@ -154,7 +154,7 @@ NESTED_ENTRY RhpWaitForGC, _TEXT, NoHandler
     bl  C_FUNC(RhpWaitForGC2)
 
     ld.d  $a2,$sp, OFFSETOF__PInvokeTransitionFrame__m_Flags
-    andi  $t8, $a2, PTFF_THREAD_ABORT_BIT
+    bstrpick.d  $t8, $a2, PTFF_THREAD_ABORT_BIT, PTFF_THREAD_ABORT_BIT
```
Comment (shushanhf) [inline src/coreclr/nativeaot/Runtime/loongarch64/GcProbe.S:157]: Please don't modify here.

File: src/coreclr/nativeaot/Runtime/unix/unixasmmacrosloongarch64.inc [hunk 3]
```diff
@@ -319,7 +319,7 @@ C_FUNC(\Name):
 #define TSF_DoNotTriggerGc              0x10
 #define TSF_SuppressGcStress__OR__TSF_DoNotTriggerGC 0x18
 
-// Bit position for the flags above, to be used with andi+beq/bne instructions
+// Bit position for the flags above, to be used with bstrpick.d+beq/bne instructions
 #define TrapThreadsFlags_AbortInProgress_Bit 0
 #define TrapThreadsFlags_TrapThreads_Bit     1
```
Comment (shushanhf) [inline src/coreclr/nativeaot/Runtime/unix/unixasmmacrosloongarch64.inc:324]:
```suggestion
// Bit position for the flags above, to be used with andi+beq/bne instructions
#define TrapThreadsFlags_AbortInProgress_Bit (1 << 0)
#define TrapThreadsFlags_TrapThreads_Bit     (1 << 1)
```

File: src/coreclr/nativeaot/Runtime/loongarch64/GcProbe.S [hunk 4]
```diff
@@ -134,7 +134,7 @@ NESTED_ENTRY RhpGcProbeHijack, _TEXT, NoHandler
     FixupHijackedCallstack
 
     PREPARE_EXTERNAL_VAR_INDIRECT_W RhpTrapThreads, $a3
-    andi  $t8, $a3, TrapThreadsFlags_TrapThreads_Bit
+    bstrpick.d  $t8, $a3, TrapThreadsFlags_TrapThreads_Bit, TrapThreadsFlags_TrapThreads_Bit
```
Comment (shushanhf) [inline src/coreclr/nativeaot/Runtime/loongarch64/GcProbe.S:137]: Please don't modify here.

**Review by shushanhf — APPROVED** [review]
LGTM
Thanks

**Review by jkotas — APPROVED** [review]

**Pull request merged**